### DISCUSSION TITLE
Is it anti-pattern to nest presented sheets?

### DISCUSSION THREAD

**Discussion by taskcruncher** [pointfreeco/swift-composable-architecture#2683] [Q&A] [answered]
Using the observation-beta, I am getting unexpected behavior attempting to present a sheet from within a presented sheet. 

**Is nesting presented sheets anti-pattern?**  Or known to trigger SwiftUI bugs?

**Comment by mbrandonw** [comment]
Hi @taskcruncher, it should be perfectly fine to present sheets inside sheets. Can you provide a minimal project that demonstrates the behavior you are seeing?

**Reply by taskcruncher** [reply, marked as answer]
TY: your response prompted me to dig deeper. The unexpected behavior was the result of using the TCA 1.5 sheet modifier `.sheet(store:)`, instead of TCA 1.6 `.sheet(item:)`. (This change is covered in the 1.6 / 1.7 migration [guide](https://github.com/pointfreeco/swift-composable-architecture/blob/observation-beta/Sources/ComposableArchitecture/Documentation.docc/Articles/MigrationGuides/MigratingTo1.7.md#replacing-navigation-view-modifiers-with-swiftui-modifiers))
